### PR TITLE
Migrate `databricks_cluster_policy` to Go SDK

### LINKED ISSUE #2247: [FEATURE] Add `description` field in `databricks_cluster_policy` Resource
### Use-cases

The Cluster Policy resource is missing the `description` field.

According to https://docs.databricks.com/api-explorer/workspace/clusterpolicies/create, I am expecting to set the description of the cluster policy. However, with the current implementation (1.14.3) it is not possible to do it.

### Proposal

The first proposal is to add only the `description` field to the `databricks_cluster_policy` Resource. This will minimize the implementation effort.

However, since Cluster Policies are in GA since April 7 (https://www.databricks.com/blog/2023/04/07/announcing-general-availability-cluster-policies.html), you should update the `databricks_cluster_policy` Resource with all the missing fields to allow the overrides of Policy Families definitions.

Thanks

### PR DESCRIPTION
**Pull request by nkvuong**
## Changes

Migrate the cluster policy service to go SDK.
Closes #2247

## Tests

- [x] `make test` run locally
- [x] relevant change in `docs/` folder
- [x] covered with integration tests in `internal/acceptance`
- [x] relevant acceptance tests are passing
- [x] using Go SDK

### CODE REVIEW

File: docs/resources/cluster_policy.md [hunk 1]
```diff
@@ -98,8 +98,11 @@ module "engineering_compute_policy" {
 The following arguments are required:
 
 * `name` - (Required) Cluster policy name. This must be unique. Length must be between 1 and 100 characters.
-* `definition` - (Required) Policy definition: JSON document expressed in [Databricks Policy Definition Language](https://docs.databricks.com/administration-guide/clusters/policies.html#cluster-policy-definition).
+* `description` - Additional human-readable description of the cluster policy.
```
Comment (alexott) [inline docs/resources/cluster_policy.md:101]: should we mark it as optional?

**Review by alexott — COMMENTED** [review]
looks good, but there are some minor comments

File: docs/resources/cluster_policy.md (hunk 1 above)
Comment (nkvuong) [reply docs/resources/cluster_policy.md]: done

File: policies/resource_cluster_policy.go [hunk 2]
```diff
@@ -3,165 +3,79 @@ package policies
...
+			resp, err := w.ClusterPolicies.GetByPolicyId(ctx, d.Id())
 			if err != nil {
 				return err
 			}
-			d.Set("name", clusterPolicy.Name)
-			d.Set("definition", clusterPolicy.Definition)
-			d.Set("policy_id", clusterPolicy.PolicyID)
-			d.SetId(clusterPolicy.PolicyID)
-			d.Set("max_clusters_per_user", clusterPolicy.MaxClustersPerUser)
+			common.StructToData(resp, s, d)
```
[the remark this answers is not in the document]
Comment (nkvuong) [reply policies/resource_cluster_policy.go]: done

File: policies/resource_cluster_policy_test.go [hunk 3]
```diff
@@ -117,6 +118,35 @@ func TestResourceClusterPolicyCreate(t *testing.T) {
...
+				ExpectedRequest: clusterpolicies.CreatePolicy{
+					Name:               "Dummy",
+					Definition:         "{\"spark_conf.foo\": {\"type\": \"fixed\", \"value\": \"bar\"}}",
+					PolicyFamilyId:     "Test",
+					MaxClustersPerUser: 3,
+				},
+				Response: clusterpolicies.CreatePolicyResponse{
+					PolicyId: "abc",
+				},
+			},
```
[the remark this answers is not in the document]
Comment (nkvuong) [reply policies/resource_cluster_policy_test.go]: you are correct, this was not needed - removed and tests are passing

**Review by alexott — APPROVED** [review]
lgtm

File: docs/resources/cluster_policy.md [hunk 4]
```diff
@@ -98,8 +98,11 @@ module "engineering_compute_policy" {
 The following arguments are required:
 
 * `name` - (Required) Cluster policy name. This must be unique. Length must be between 1 and 100 characters.
-* `definition` - (Required) Policy definition: JSON document expressed in [Databricks Policy Definition Language](https://docs.databricks.com/administration-guide/clusters/policies.html#cluster-policy-definition).
+* `description` - (Optional) Additional human-readable description of the cluster policy.
+* `definition` - Policy definition: JSON document expressed in [Databricks Policy Definition Language](https://docs.databricks.com/administration-guide/clusters/policies.html#cluster-policy-definition). Cannot be used with `policy_family_id`
```
Comment (mgyucht) [inline docs/resources/cluster_policy.md:102]: Should this be labeled as `Optional` explicitly, like `policy_family_id`?

File: docs/resources/cluster_policy.md [hunk 5]
```diff
@@ -98,8 +98,11 @@ module "engineering_compute_policy" {
...
 * `max_clusters_per_user` - (Optional, integer) Maximum number of clusters allowed per user. When omitted, there is no limit. If specified, value must be greater than zero.
+* `policy_family_definition_overrides`(Optional) Policy definition JSON document expressed in Databricks Policy Definition Language. The JSON document must be passed as a string and cannot be embedded in the requests. You can use this to customize the policy definition inherited from the policy family. Policy rules specified here are merged into the inherited policy definition.
+* `policy_family_id` (Optional) ID of the policy family. The cluster policy's policy definition inherits the policy family's policy definition. Cannot be used with `definition`. Use `policy_family_definition_overrides` instead to customize the policy definition.
```
Comment (mgyucht) [inline docs/resources/cluster_policy.md:105]: formatting nits: make sure ` - ` comes between the field name and `(Optional)`

File: docs/resources/cluster_policy.md [hunk 6]
```diff
@@ -98,8 +98,11 @@ module "engineering_compute_policy" {
...
-* `definition` - (Required) Policy definition: JSON document expressed in [Databricks Policy Definition Language](https://docs.databricks.com/administration-guide/clusters/policies.html#cluster-policy-definition).
+* `description` - (Optional) Additional human-readable description of the cluster policy.
+* `definition` - Policy definition: JSON document expressed in [Databricks Policy Definition Language](https://docs.databricks.com/administration-guide/clusters/policies.html#cluster-policy-definition). Cannot be used with `policy_family_id`
 * `max_clusters_per_user` - (Optional, integer) Maximum number of clusters allowed per user. When omitted, there is no limit. If specified, value must be greater than zero.
+* `policy_family_definition_overrides`(Optional) Policy definition JSON document expressed in Databricks Policy Definition Language. The JSON document must be passed as a string and cannot be embedded in the requests. You can use this to customize the policy definition inherited from the policy family. Policy rules specified here are merged into the inherited policy definition.
```
Comment (mgyucht) [inline docs/resources/cluster_policy.md:104]: I understand what it means about the JSON document being passed in as a string, but what does "The JSON document ... cannot be embedded in the requests" mean?

**Review by mgyucht — APPROVED** [review]
Thank you for doing this! It would be good to get @nfx's review on this as well possibly. Just a couple of nits on the documentation.

**Pull request merged**